core/src/hir/lowering.rs

```diff
@@ -1064,10 +1096,10 @@ impl<'ast> LoweringContext<'ast> {
         ltl: &mut impl LifetimeLowerer,
         in_path: &ast::Path,
     ) -> Result<Param, ()> {
-        let name = self.lower_ident(&param.name, "param name");
-        let ty = self.lower_type::<InputOnly>(&param.ty, ltl, in_path);
+        let name = self.lower_ident(&param.name, "param name")?;
```

Comment:
oh, this change to the `?` shouldn't have happened, we lazily `?` so that compilation can proceed as far as possible before errors cause it to fail